src/mongo/db/storage/storage_engine_init.cpp

```diff
@@ -249,9 +249,30 @@ StorageEngine::LastShutdownState initializeStorageEngine(OperationContext* opCtx
         }
     });
 
-    encryptionGlobalParams.kmipKeyIds.encryption = encryptionGlobalParams.kmipKeyIdentifier;
-    encryptionGlobalParams.kmipKeyIds.decryption =
-        metadata ? metadata->getKmipMasterKeyId() : encryptionGlobalParams.kmipKeyIdentifier;
+    if (!encryptionGlobalParams.kmipServerName.empty()) {
+        const std::string& configuredId = metadata ? metadata->getKmipMasterKeyId() : "";
+        const std::string& providedId = encryptionGlobalParams.kmipKeyIdentifier;
+        if (encryptionGlobalParams.kmipRotateMasterKey && configuredId.empty()) {
+            LOGV2_FATAL_NOTRACE(
+                29112,
+                "The system is not configured with a KMIP-managed key but the command line opiton "
+                "or the configuration file asks to rotate the KMIP master key.");
+        }
+        bool keyIdMisconfig = !encryptionGlobalParams.kmipRotateMasterKey &&
+            !configuredId.empty() && !providedId.empty() && configuredId != providedId;
+        if (keyIdMisconfig) {
+            LOGV2_FATAL_NOTRACE(
+                29113,
+                "The provided (via the command line option or the configuration file) KMIP "
+                "keyIdentifier is not equal to that the system is already configured with. "
+                "Don't provide keyIdentifier or rotate master key.",
```

Comment:
This line is concise but not very clear. There are 3 options in this situation:
- provide matching value with --kmipKeyIdentifier
- omit --kmipKeyIdentifier
- provide --kmipRotateMasterKey along with non-matching value of --kmipKeyIdentifier (if it was intended to change key identifier)

Not sure how to improve this error message.